drucker/drucker-dashboard
# evaluate model API

## What is this PR for?

Implement API to evaluate model:
(ref of protocol: https://github.com/drucker/drucker-grpc-proto/pull/10 )
(This PR is just for `EvaluateModel` protocol. PR for `EvaluationResult` protocol will be the next)

## This PR includes

- implement ApiEvaluate class to request evaluation to drucker_worker
- save evaluation data path to db (create `Evaluation` model class)
- try using protobuf3-to-dict library instead of `ProtobufUtil.serialize_to_object`

## What type of PR is it?

Feature

## What is the issue?

N/A

## How should this be tested?

- start drucker server based on https://github.com/drucker/drucker/pull/11 and https://github.com/drucker/drucker-example/pull/5
- start drucker-dashboard backend server
- request to evaluation API with the following script

```
import requests

app_id = 1  # depends on your local data
service_id = 1  # depends on your local data
url = 'http://localhost:18080/api/applications/{}/services/{}/evaluate'.format(app_id, service_id)
res = requests.post(url, files={'file': open('dummy.txt')})
print(res.text)  # check API response
``` 

(dummy.txt is like below)

```
0,0,5,0,0,0,0,0,0,2,0,13,0,0,0,0,0,12,0,4,0,10,0,0,0,0,0,0,0,17,0,0,0,0,0,1,0,0,0,0,8,1,12,0,0,0,0,7,0,6,11,0,0,1,0,13,0,8,10,9,0,0,0,16,13
1,0,0,0,0,0,11,13,2,0,0,11,12,0,2,15,8,0,18,5,0,11,6,5,0,5,0,0,0,16,6,0,16,18,0,0,9,0,0,0,16,0,0,0,0,0,4,0,0,17,0,9,10,4,5,15,6,0,7,1,4,0,0,0,10
```
